frontend/src/middleware.ts

```diff
@@ -0,0 +1,27 @@
+import { getToken } from 'next-auth/jwt'
+import { NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
+import { validateToken } from './infra/lib/validateToken'
+
+const protectedRoutes = ['/tasks']
```

Comment:
Eventually, all routes will be protected except for maybe a generic landing page. Nothing to change here - just making that note.